internal/archive/zip/zip.go

```diff
@@ -6,21 +6,25 @@ import (
 	"os"
 	"path"
 	"path/filepath"
+	"strings"
+
+	"github.com/saucelabs/saucectl/internal/sauceignore"
 )
 
 // Writer is a wrapper around zip.Writer and implements zip archiving for archive.Writer.
 type Writer struct {
 	W *zip.Writer
+	M sauceignore.Matcher
```

Comment:
Should this interface be defined in `zip` instead of `sauceignore` package ?

Reply:
For docker, I will add a `tar` package inside an `archive` package, so we can rather add `sauceignore` as `archive` subfolder, but I would leave `sauceignore` as a separate package.

Reply:
I was meaning it bothers me that `zip` has a dependency on `sauceignore`.
`sauceignore` implements a `Matcher` used by `zip`.
This matcher could also be something not related to `sauceignore`.

Reply:
Try to think about it as a dependency. If this dependency exists then we can skip some folders and files. `archive` package right now is how developers of `golang` language structure their packages. `archive/archive.go` is a basic interface and `archive/zip/zip.go` implements this interface. I would not mix the logic of archiving and applying sauceignore.